runtime/lua/vim/secure.lua

```diff
@@ -86,21 +109,79 @@ function M.read(path)
     trust[fullpath] = hash
   end
 
-  do
-    local f, err = io.open(vim.fn.stdpath('state') .. '/trust', 'w')
-    if not f then
-      error(err)
+  write_trust(trust)
+
+  return contents
+end
+
+--- Update the trust status of file at {path} in the trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+---@param opts (table):
+---    - action (string): "allow" to add a file to the trust database and trust it,
+---      "deny" to add a file to the trust database and deny it,
+---      "remove" to remove file from the trust database
+---    - path (string|nil): Path to a file to update. If nil, must provide bufnr
+---    - bufnr (number|nil): Buffer number to update. If nil, must provide path.
+---      For "allow", must provide bufnr.
```

Comment:
```suggestion
---    - path (string|nil): Path to a file to update. Mutually exclusive with {bufnr}. Cannot be used when {action} is "allow".
---    - bufnr (number|nil): Buffer number to update. Mutually exclusive with {path}.
```

Reply:
Done in d3d5654ca469c73ed196ac902a9e1eaadf4d0ec1